presto-main/src/main/java/io/prestosql/server/CoordinatorModule.java

```diff
@@ -210,6 +211,7 @@ protected void setup(Binder binder)
         jaxrsBinder(binder).bind(QueryStateInfoResource.class);
         jaxrsBinder(binder).bind(ResourceGroupStateInfoResource.class);
         binder.bind(QueryIdGenerator.class).in(Scopes.SINGLETON);
+        binder.bind(StatementAccessControl.class).toProvider(StatementAccessControlProvider.class).in(Scopes.SINGLETON);
```

Comment:
`StatementAccessControl` should be modeled similarly as `EventListenerFactory` and `EventListener` (it should be exportable by `Plugin`). See corresponding classes:
```
* Plugin
* EventListener
* EventListenerFactory
* EventListenerManager
* EventListenerModule
* PluginManager
```

Reply:
OK. Shall modify the same

Reply:
Created the following and made the BT plugin adhere to current presto-spi ```Plugin``` interface
```
StatementAccessControl
StatementAccessControlFactory
StatementAccessControlManager
```